Automattic/wp-calypso
# Staging Sites: Add staging badge to SMP and site switcher

- Close https://github.com/Automattic/dotcom-forge/issues/1885

## Proposed Changes

* This PR adds a site badge to SMP (both grid and row view) and to the site switcher.

## Testing Instructions

* Create a staging site from **Settings > Hosting Configuration** of an Atomic site
* Go to `/sites`
* Check the row and grid view
* Click on the title of the staging site from SMP to open the dashboard
* Check the staging badge in the sites' list and in the dashboard (see the screenshots below)

## Screenshots

**Row View**: 

<img width="1148" alt="Screenshot 2023-03-08 at 11 12 38 AM" src="https://user-images.githubusercontent.com/25575134/223685440-7245efcc-6073-4ee5-a0d2-178aef3bd33a.png">

**Grid View**:

<img width="1143" alt="Screenshot 2023-03-08 at 11 12 24 AM" src="https://user-images.githubusercontent.com/25575134/223685564-a8e9552b-4475-4a75-a254-e33804938c85.png">

**Site Switcher**:

<img width="630" alt="Screenshot 2023-03-08 at 11 14 35 AM" src="https://user-images.githubusercontent.com/25575134/223685856-f63b6c5e-fab7-4f33-a238-0068b8155f2f.png">

<img width="549" alt="Screenshot 2023-03-08 at 11 14 15 AM" src="https://user-images.githubusercontent.com/25575134/223685880-5036f367-98e1-4b71-b403-7bc280d4a2ac.png">

## Pre-merge Checklist
